libcaf_net/caf/net/http/server_factory.hpp

```diff
@@ -219,7 +219,7 @@ class server_factory
     routes.push_back(make_route([producer](responder& res) {
       if (!producer->push(std::move(res).to_request())) {
         auto err = make_error(sec::runtime_error, "flow disconnected");
-        res.router()->shutdown(err);
+        res.router()->shutdown(err); // NOLINT(bugprone-use-after-move)
```

Comment:
Can't we drop the move above? This actually looks suspicious to me.

Reply:
We depend on the contract that "If the `producer::push` fails it won't break the rvalue argument". It looks like something hard to enforce, but I left the code as is.